### ISSUE TITLE
Different error reporting between backends

### ISSUE THREAD

**Issue by mwillerich** [lostisland/faraday#584]
Hi, I'm setting up a Faraday connection like so:
```
conn = Faraday.new do |c|
  c.use FaradayMiddleware::FollowRedirects, limit: 5
  c.use Faraday::Response::RaiseError       # raise exceptions on 40x, 50x responses
  c.use Faraday::Adapter::NetHttp
end
```
and then run a list of urls on the connection to check whether they come back ok. If they don't, I want to capture why the request failed.
```
begin
  response = conn.get URI(url)
  #check response.status and do stuff
rescue Faraday::SSLError
  #do stuff
rescue Faraday::ClientError
  #do stuff
end
```

When I use `Faraday::Adapter::NetHttp`, I get some SSLErrors and a ClientError (a few SSLErrors that Chrome wouldn't care about, but it's generally correct), but when I use `Faraday::Adapter::Typhoeus`, I only get ClientErrors. Is it possible there's a bug in [this routine](https://github.com/lostisland/faraday/blob/80c0e66102168939df04682573bc267e089a43ae/lib/faraday/adapter/typhoeus.rb#L59-L68)?

A related question to this: If I run Typhoeus in parallel, how can I act on either SSLErrors or ClientErrors? They don't seem to be raised. Happy to open this in a separate issue or address this in a different forum, though.
